java/src/main/java/io/cucumber/query/Query.java

```diff
@@ -135,10 +141,28 @@ public List<TestStep> findAllTestSteps() {
                 .collect(toList());
     }
 
+    public List<Attachment> findAttachmentsBy(TestStepFinished testStepFinished) {
+        requireNonNull(testStepFinished);
+        return this.attachmentsByTestCaseStartedId.getOrDefault(testStepFinished.getTestCaseStartedId(), emptyList()).stream()
```

Comment:
nitpick: Using `this` outside of a constructor  isn't common. It is mostly used for disambiguation between method arguments and class members.